exodus_lambda/functions/verify_source.py

```diff
@@ -0,0 +1,56 @@
+import json
+import os
+import subprocess
+
+import boto3
+
+
+def lambda_handler(event, context):
+    """Verifies the commit delivered to the pipeline by GitHub has a
+    tag signed by a trusted entity.
+
+    Receives the commit ID from the pipeline via UserParameters.
+
+    This function must be manually deployed prior to pipeline creation.
+    """
+    # pylint: disable=unused-argument
+
+    job_id = event["CodePipeline.job"]["id"]
+    job_data = event["CodePipeline.job"]["data"]
+    job_params = json.loads(
+        job_data["actionConfiguration"]["configuration"]["UserParameters"]
+    )
+
+    pipeline_client = boto3.client("codepipeline")
+
+    try:
+        # Import whitelisted GPG keys
+        pub_keys = "".join(
+            pub_key for _, pub_key in job_params["whitelist"].items()
+        )
+        subprocess.run(["gpg", "--import", pub_keys], check=True)
+
+        # Clone exodus-lambda and checkout deploy branch
+        repo_url = "https://github.com/release-engineering/exodus-lambda.git"
+        subprocess.run(["git", "clone", repo_url, "-b", "deploy"], check=True)
+        os.chdir("exodus-lambda")
```

Comment:
Just a quick note that if you do start accepting the repo name as a parameter, this part would ideally not be hardcoded any more.

I would suggest using the form of git clone where you specify the output directory instead of letting git choose for you, e.g. "git clone <repo_url> src -b deploy" and then you know it's always cloned to "src" regardless of the repo's name.